site/src/components/UserAutocomplete/UserAutocomplete.tsx

```diff
@@ -0,0 +1,126 @@
+import CircularProgress from "@material-ui/core/CircularProgress"
+import { makeStyles } from "@material-ui/core/styles"
+import TextField from "@material-ui/core/TextField"
+import Autocomplete from "@material-ui/lab/Autocomplete"
+import { useMachine } from "@xstate/react"
+import { User } from "api/typesGenerated"
+import { AvatarData } from "components/AvatarData/AvatarData"
+import debounce from "just-debounce-it"
+import { ChangeEvent, useEffect, useState } from "react"
+import { searchUserMachine } from "xServices/users/searchUserXService"
+
+export type UserAutocompleteProps = {
+  value?: User
+  onChange: (user: User | null) => void
+}
+
+export const UserAutocomplete: React.FC<UserAutocompleteProps> = ({ value, onChange }) => {
+  const styles = useStyles()
+  const [isAutocompleteOpen, setIsAutocompleteOpen] = useState(false)
+  const [searchState, sendSearch] = useMachine(searchUserMachine)
+  const { searchResults } = searchState.context
+  const [selectedValue, setSelectedValue] = useState<User | null>(value || null)
+
+  // seed list of options on the first page load
+  useEffect(() => {
+    const query = value ? value.email : ""
+    sendSearch("SEARCH", { query })
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
```

Comment:
We have some users in v1 with 1000 ~ 2,500 users so loading them at the beginning can be heavy. I think we could only load this when the user start to type in the search box + debouce(to avoid extra loads during typing). What do you think?

Reply:
@BrunoQuaresma good point! Your solution sounds right. So we can get this out there (@kylecarbs wants to use it) how about we do something like:
````
  useEffect(() => {
    if (value) {
      sendSearch("SEARCH", { query: value.email })
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [])
````

for now?

Reply:
Sounds good!